client/my-sites/domains/components/domain-and-plan-package/navigation/index.jsx

```diff
@@ -1,13 +1,18 @@
 import { Button, Gridicon } from '@automattic/components';
 import { useTranslate } from 'i18n-calypso';
+import page from 'page';
 
 import './style.scss';
 
 export default function DomainAndPlanPackageNavigation( props ) {
 	const translate = useTranslate();
 
 	const goBack = () => {
-		window.history.go( -1 );
+		if ( props.goBackLink ) {
+			page( props.goBackLink );
+		} else {
+			window.history.go( -1 );
+		}
```

Comment:
Another approach is to write the links that we want to go back here, for example something like this.
```
switch (step) {
  case 1:
     // redirect somewhere
   case 2:
    // redirect somewhere
   default:
       window.history.go(-1);
}
```

but I think that's more of a personal preference.